spec/std/time/time_spec.cr

```diff
@@ -13,18 +13,38 @@ describe Time do
     t1.month.should eq(2)
     t1.day.should eq(25)
 
-    t2 = Time.new 2002, 2, 25, 15, 25, 13, 8
+    t2 = Time.new 2002, 2, 25, 15, 25, 13, nanosecond: 8
     t2.year.should eq(2002)
     t2.month.should eq(2)
     t2.day.should eq(25)
     t2.hour.should eq(15)
     t2.minute.should eq(25)
     t2.second.should eq(13)
     t2.nanosecond.should eq(8)
+
+    t3 = Time.new 2002, 2, 25, 15, 25, 13, millisecond: 8
+    t3.year.should eq(2002)
+    t3.month.should eq(2)
+    t3.day.should eq(25)
+    t3.hour.should eq(15)
+    t3.minute.should eq(25)
+    t3.second.should eq(13)
+    t3.nanosecond.should eq(8_000_000)
+  end
+
+  it "initialize max with milliseconds" do
+    time = Time.new(9999, 12, 31, 23, 59, 59, millisecond: 999)
```

Comment:
Is this correct? or should we use `999.999_999` so that it expands to `999_999_999` nanoseconds?

Reply:
`999.999_999` won't pass the `0 <= millisecond <= 999`
I think the issue is the name of the spec.

Reply:
we could then use `0 <= millisecond < 1_000` ?

Reply:
I think they are restricted to be integer since the `initialize(*, @seconds : Int64, @nanoseconds : Int32, @kind : Kind)` is called at the end of the day

Reply:
We could add `.to_i` before calling the `Int32` restricted constructor. I think that would work OK.

Reply:
I prefer them integer actually. All of them. But that's just my opinion 🤷‍♂️